### ISSUE TITLE
Integrating JARVIS with dmenu

### ISSUE THREAD

**Issue by Icyphox** [swapagarwal/JARVIS-on-Messenger#119]
I think I have a great idea to integrate JARVIS with the pretty popular program launcher dmenu.
It's possible to write a simple script to point to the URL to make the query, but the only issue is the output text from the URL isn't pretty.

So... what do you say?

**Comment by swapagarwal** [comment]
Sorry, I'm not familiar with it. Can you provide more details on the response format that you would prefer?
Also, it'd be helpful if you could provide a sample workflow of what you are trying to do with dmenu. 😄

**Comment by Icyphox** [comment]
Okay dmenu is a program launcher for Linux, mostly used in tiling window managers. Anyway, I kinda forgot about this, but my idea was to run Jarvis through dmenu, and add it as an extended plugin, which behaves like a personal assistant. 

Nevermind, you can close this issue. Thanks anyway.

**Comment by swapagarwal** [comment]
So something like exposing jarvis as an api, sounds cool!
Let's keep it open for further discussion. Are you familiar with integrations in dmenu?

**Comment by Icyphox** [comment]
Yeah, dmenu plugins are simple Python scripts. Here look at [this](https://github.com/markjones112358/dmenu-extended-plugins). Should give you an idea of what I'm talking about.

**Comment by swapagarwal** [comment]
Okay, this would require jarvis to be hosted as a service at some public endpoint that you can hit. Open to suggestions...

**Comment by Icyphox** [comment]
Yeah, that's probably required. Is it possible? Maybe for starters, run it on a remote VPS?

**Comment by swapagarwal** [comment]
@Icyphox #151 seems to solve this, i.e. you can host jarvis on your local machine itself!

**Comment by Icyphox** [comment]
@swapagarwal Well, I created that pull, because of this :)

**Comment by swapagarwal** [comment]
@Icyphox If you want to deploy to a public endpoint, you can use Heroku. Visit this link to host your own instance: https://heroku.com/deploy?template=https://github.com/swapagarwal/JARVIS-on-Messenger

**Comment by Icyphox** [comment]
Oh okay, that's great.
